### PR TITLE
host-local: detect missing subnet and error out

### PR DESCRIPTION
**Pull request by steveeJ**
Fixes https://github.com/coreos/rkt/issues/1707

### CODE REVIEW

File: plugins/ipam/host-local/allocator.go [hunk 1]
```diff
@@ -144,6 +144,9 @@ func (a *IPAllocator) Release(id string) error {
 func networkRange(ipnet *net.IPNet) (net.IP, net.IP, error) {
 	ip := ipnet.IP.To4()
 	if ip == nil {
+		if ipnet.IP == nil {
+			return nil, nil, fmt.Errorf("missing field %q in IPAM configuration", "subnet")
+		}
```
Comment (alban) [inline plugins/ipam/host-local/allocator.go]: I would move the 3 new lines at the top of the function.

Comment (alban) [inline plugins/ipam/host-local/allocator.go]: Except that, LGTM, but I am not a maintainer.

**Comment by alban** [comment]
LGTM, but I am not a maintainer.

**Comment by iaguis** [comment]
ping?

**Comment by jonboulle** [comment]
LGTM, @steveeJ is a maintainer so he can land this

**Pull request merged**